From fccbc523ee9ef029bf12237ad7926ea75f6fa85c Mon Sep 17 00:00:00 2001
From: tgeisler <tgeisler@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 4 Oct 2016 15:04:44 +0000
Subject: [PATCH 0799/1076] MANTIS 0043032 DoorDevice: DoorDev more
 functionality IDL extension FRAUD_DOOR_FORCED, FRAUD_JUMP_OVER, more error
 info, new stat normal pass

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1096 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/doordev.idl             | 38 +++++++++++++++----
 1 file changed, 30 insertions(+), 8 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/doordev.idl b/SelfServiceCommon/Massai/idl/Massai/doordev.idl
index 4d5b1fb8..f663d971 100644
--- a/SelfServiceCommon/Massai/idl/Massai/doordev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/doordev.idl
@@ -6,7 +6,7 @@
  * Contains module doordev, which is the platform-internal interface
  * definition to test the common use gate's door device component.
  *
- * @version 1.7
+ * @version 1.8
  *
  * @author Sven Bornemann
  *
@@ -87,7 +87,8 @@ module doordev
         STAT_GT_FRAUD,                  /* Fraud detected */
         STAT_GT_PANIC_OPEN,             /* Gate is in panic open/firealarm mode */
         STAT_GT_DOOR_BLOCKED,           /* Door is blocked*/
-        STAT_GT_REVERSE_PASS            /* Door is handling a reverse pass that was automatically or by button triggered*/
+        STAT_GT_REVERSE_PASS,           /* Door is handling a reverse pass that was triggered automatically or by button*/
+        STAT_GT_NORMAL_PASS             /* Door is handling a normal pass that was triggered automatically or by a remote switch*/
     };
     
     enum DOOR_STAT
@@ -114,6 +115,24 @@ module doordev
         ERROR_OOS,                      /* out of service */
         ERROR_PWR                       /* power failure */
     };
+
+    enum ERROR_INFO_TYPE
+    {
+        ERROR_HW_SENSOR,                /* specific sensor failure information - value: failed sensor id(number) */
+        ERROR_HW_MOTOR_DRIVER,          /* specific motor driver failure information - value: motor driver fault code  */
+        ERROR_HW_ENCODER,               /* specific encoder failure information - value: encoder fault code  */
+        ERROR_HW_RELAY,                 /* specific relay failure information - value: relay fault code */
+        ERROR_HW_SAFETY_DEVICE          /* specific (Neplite) safety device failure information - value: safety device fault code */
+    };
+    
+    struct fullErrorInfo
+    {
+        ERROR_INFO_TYPE error_info;     /**< type of specific error information */
+        long            value;          /**< error specific value ( normally an error code or an id ) */
+    };
+
+    /**< List full error information tuple */
+    typedef sequence<fullErrorInfo> fullErrorInfoList;
     
     enum FRAUD_TYPE
     {
@@ -126,7 +145,9 @@ module doordev
         FRAUD_TAIL_GATING,              /* Tail gating has been detected */
         FRAUD_WRONG_WAY,                /* Wrong Way has been detected */
         FRAUD_CRAWL_UNDER,              /* A person crawled under the flaps */
-        FRAUD_TAMPER_DETECTED           /* Any cabinet door is opened */
+        FRAUD_TAMPER_DETECTED,          /* Any cabinet door is opened */
+        FRAUD_DOOR_FORCED,              /* A closed door is forcefully opened */
+        FRAUD_JUMP_OVER                 /* A person jumped over the flaps */
     };
     
     enum BUTTON_TYPE
@@ -144,11 +165,12 @@ module doordev
     /**< hardware status of the gate */
     struct hwStatusType
     {
-        GATE_STAT      gate_state;
-        ERROR_TYPE     error;
-        DOOR_STAT      door_state;
-        FRAUD_TYPE     fraud;
-        DOOR_INFO      door_information;
+        GATE_STAT           gate_state;
+        ERROR_TYPE          error;
+        DOOR_STAT           door_state;
+        FRAUD_TYPE          fraud;
+        DOOR_INFO           door_information;
+        fullErrorInfoList   error_informations;
     };
     
     /**< Door device capabilities */
-- 
2.41.0.windows.1

